### PR TITLE
Implement `updateUser` command

### LINKED ISSUE #1496: Implement `updateUser` command
### What should be done?

With passwords being stored, we should implement the basic [`updateUser ` command](https://www.mongodb.com/docs/manual/reference/command/updateUser/).

Also, in this issue, we should update _all_ integration tests to use `createUser`, `dropUser`, `dropAllUsersFromDatabase`, `usersInfo` commands instead of using `find`/`insert`/`delete`/etc on the `admin.system.users` collection.

### Where?

https://github.com/FerretDB/FerretDB/tree/main/internal/handler

### Definition of Done

- handler updated;
- integration/compatibility tests added/updated;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by henvic**
# Description

Initial implementation of the updateUser command.

Closes #1496.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handler/msg_updateuser.go [hunk 1]
```diff
@@ -29,6 +38,201 @@ func (h *Handler) MsgUpdateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1496
-	return nil, common.Unimplemented(document, document.Command())
+	dbName, err := common.GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, lazyerrors.Error(err)
```
Comment (rumyantseva) [inline internal/handler/msg_updateuser.go:43]: Minor comment for here and below: I think errors returned after `GetRequiredParam` don't have to be `lazyerrors`, because `GetRequiredParam` returns a user-friendly error that we just need to show as is.

File: integration/users/update_user_test.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+			updatePayload: bson.D{
+				{"updateUser", "missing_fields"},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "Must specify at least one field to update in updateUser",
+			},
+		},
+		"UserNotFound": {
```
Comment (rumyantseva) [inline integration/users/update_user_test.go:63]: Do we need a separate test where username is set as an empty string when update is called? Or is this one enough for this case too?

File: internal/handler/msg_updateuser.go [hunk 3]
```diff
@@ -29,6 +38,201 @@ func (h *Handler) MsgUpdateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
...
+
+		var matches bool
+		matches, err = common.FilterDocument(v, filter)
+
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		if matches {
+			saved = v
```
Comment (rumyantseva) [inline internal/handler/msg_updateuser.go:199]: do also we need to close the iterator or exit from the loop or something?

**Review by rumyantseva — COMMENTED** [review]
Looks great to me overall, I especially like how test cases cover various scenarios! I added a few minor questions/comments (feel free to disagree).

File: integration/users/update_user_test.go (hunk 2 above)
Comment (henvic) [reply integration/users/update_user_test.go:63]: Good idea!

File: internal/handler/msg_updateuser.go (hunk 3 above)
Comment (henvic) [reply internal/handler/msg_updateuser.go:199]: Iterator is deferred closed on preceding line `qr.Iter.Close()`.

This loop should always contain a maximum 1 item, but I'm adding a `break` just to make it more explicit.

Comment (rumyantseva) [reply internal/handler/msg_updateuser.go:199]: Hmm, could you help me understand why this loop should always contain a maximum 1 item? Doesn't it contain all the users?

Comment (henvic) [reply internal/handler/msg_updateuser.go:199]: Oops! You're correct. Please disregard my comment.

It should contain at maximum one item, but we're not filtering at the database layer at this moment.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by noisersup — APPROVED** [review]
LGTM
